fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,67 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclen := *(*uint16)(unsafe.Pointer(&buf[16]))
+		ino := *(*uint64)(unsafe.Pointer(&buf[0]))
+
+		if int(reclen) > len(buf) {
+			return count
+		}
+		buf = buf[reclen:]
+		if ino == 0 {
+			continue
+		}
+		count++
+	}
+	return count
+}
+
+func getdents64(fd int, buf []byte) (n int, err int) {
+	var _p0 unsafe.Pointer
+	_p0 = unsafe.Pointer(&buf[0])
+	r0, _, errno := syscall.Syscall(syscall.SYS_GETDENTS64, uintptr(fd), uintptr(_p0), uintptr(len(buf)))
+	n = int(r0)
+	err = int(errno)
+	return
+}
+
+// ReadDirCount is similar to ReadDirNames() and then counting with len() but
+// it is optimized to avoid parsing the entries
+func (realFS) ReadDirCount(dir string) (int, error) {
+	buf := make([]byte, 4096)
```

Comment:
Have you played with this buffer size? [Golang defaults it to 4096](https://github.com/golang/go/blob/c05b06a12d005f50e4776095a60d6bd9c2c91fac/src/os/dir_unix.go#L16) but, while we are it, Scope may benefit from an even larger size (I haven't done the math though)